src/Fixer/PhpUnit/PhpUnitSetUpTearDownVisibilityFixer.php

```diff
@@ -66,11 +66,21 @@ protected function applyPhpUnitClassFix(Tokens $tokens, int $startIndex, int $en
         $counter = 0;
         $tokensAnalyzer = new TokensAnalyzer($tokens);
 
-        for ($i = $endIndex - 1; $i > $startIndex; --$i) {
+        $slicesToInsert = [];
+
+        for ($i = $startIndex; $i < $endIndex; ++$i) {
             if (2 === $counter) {
                 break; // we've seen both method we are interested in, so stop analyzing this class
             }
 
+            // skip anonymous classes insite TestCase class, which we can observe eg in setUp method
+            if ($tokens[$i]->isGivenKind(T_CLASS) && $tokensAnalyzer->isAnonymousClass($i)) {
```

Comment:
What would you say about instead of checking explicitly for anonymous class, we skip all the `{` `}` blocks?

Take a look at https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8366 - I've also removed the `isSetupOrTearDownMethod` method - it was called for every index and a new object of `TokensAnalyzer` was created there every time which - beside resource consumptions - was not necessary at all (try to remove it at `master` with the part `!$tokensAnalyzer->isLambda($index)` - all tests will pass).